GVFS/GVFS.UnitTests/Maintenance/LooseObjectStepTests.cs

```diff
@@ -0,0 +1,135 @@
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Mock.FileSystem;
+using GVFS.UnitTests.Mock.Git;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace GVFS.UnitTests.Maintenance
+{
+    [TestFixture]
+    public class LooseObjectStepTests
+    {
+        [TestCase]
+        public void LooseObjectsIgnoreTimeRestriction()
+        {
+            this.TestSetup(DateTime.Now.ToString(), out MockTracer tracer, out MockGitProcess gitProcess, out GVFSContext context);
+
+            LooseObjectsStep step = new LooseObjectsStep(context, requireCacheLock: false, ignoreTimeRestriction: true);
+            step.Execute();
+
+            tracer.StartActivityTracer.RelatedErrorEvents.Count.ShouldEqual(0);
+            tracer.StartActivityTracer.RelatedWarningEvents.Count.ShouldEqual(0);
+            List<string> commands = gitProcess.CommandsRun;
+            commands.Count.ShouldEqual(1);
+            commands[0].ShouldEqual("prune-packed -q");
+        }
+
+        [TestCase]
+        public void LooseObjectsFailTimeRestriction()
+        {
+            this.TestSetup(DateTime.Now.ToString(), out MockTracer tracer, out MockGitProcess gitProcess, out GVFSContext context);
+
+            LooseObjectsStep step = new LooseObjectsStep(context, requireCacheLock: false, ignoreTimeRestriction: false);
+            step.Execute();
+
+            tracer.StartActivityTracer.RelatedErrorEvents.Count.ShouldEqual(0);
+            tracer.StartActivityTracer.RelatedWarningEvents.Count.ShouldEqual(1);
+            List<string> commands = gitProcess.CommandsRun;
+            commands.Count.ShouldEqual(0);
+        }
+
+        [TestCase]
+        public void LooseObjectsPassTimeRestriction()
+        {
+            this.TestSetup(DateTime.Now.AddDays(-7).ToString(), out MockTracer tracer, out MockGitProcess gitProcess, out GVFSContext context);
+
+            LooseObjectsStep step = new LooseObjectsStep(context, requireCacheLock: false, ignoreTimeRestriction: false);
+            step.Execute();
+
+            tracer.StartActivityTracer.RelatedErrorEvents.Count.ShouldEqual(0);
+            tracer.StartActivityTracer.RelatedWarningEvents.Count.ShouldEqual(0);
+            List<string> commands = gitProcess.CommandsRun;
+            commands.Count.ShouldEqual(1);
+            commands[0].ShouldEqual("prune-packed -q");
+        }
+
+        [TestCase]
+        public void LooseObjectsCount()
+        {
+            this.TestSetup(DateTime.Now.AddDays(-7).ToString(), out MockTracer tracer, out MockGitProcess gitProcess, out GVFSContext context);
+
+            LooseObjectsStep step = new LooseObjectsStep(context, requireCacheLock: false, ignoreTimeRestriction: false);
+            int count = step.CountLooseObjects();
+
+            count.ShouldEqual(3);
+        }
+
+        private void TestSetup(string lastRunTime, out MockTracer tracer, out MockGitProcess gitProcess, out GVFSContext context)

```

Comment:
Instead of a lot of out-params, could we set private member variables? See `GitMaintenanceStepTests` for an example.

Reply:
@derrickstolee will the test be run concurrently?  If so private variables aren't a good idea.

Reply:
I don't think rests in the same class are run concurrently, but that may be something about MSTest and not NUnit.